engine/src/main/java/org/terasology/rendering/world/WorldRendererImpl.java

```diff
@@ -152,6 +159,21 @@
     private DisplayResolutionDependentFBOs displayResolutionDependentFBOs;
     private ShadowMapResolutionDependentFBOs shadowMapResolutionDependentFBOs;
 
+    private static RenderGraph renderGraph = new RenderGraph(); // TODO: Try making this non-static
+
+    // Required for the reflection magic that Systems use.
```

Comment:
Regarding point 1: The WorldRendererImpl constructor takes in a context and a GLBufferPool. I *can* get the first one via registry (not a good idea, I guess?), but I really don't know how to go about the second one.

Regarding point 2: On it :D